### PR TITLE
fix: fix image build bug and add some note

### PR DESCRIPTION
**Pull request by carasue**
Change-Id: I10d1cc3d1b4b38805d6db85eed7eb0e4a6f251a3

### CODE REVIEW

File: templates/docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: "3.7"
+version: "3.2"
```
Comment (Shpota) [inline templates/docker-compose.yml]: Why do you decrease the version of dicker compose?

File: templates/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,10 @@
 FROM node:12.16 AS JS_BUILD
 COPY webapp /webapp
-WORKDIR webapp
+WORKDIR /webapp
 RUN npm install && npm run build --prod
 
 FROM golang:1.13.8-alpine AS GO_BUILD
-RUN apk add build-base
+RUN  apk update && apk add --virtual build-dependencies build-base wget
```
Comment (Shpota) [inline templates/Dockerfile]: Why do you add two additional packages if `build-base` is enough? Does it not work in some cases?

What would be the benefit of using `--virtual` here?

**Comment by carasue** [comment]
This version is quite high unnecessarily thus I can not build the image unless I have the latest installed. After I googled, I found many repos has the same issues for the 3.7 version.

________________________________
发件人: Sasha Shpota <notifications@github.com>
发送时间: Monday, February 24, 2020 2:11:11 AM
收件人: Shpota/goxygen <goxygen@noreply.github.com>
抄送: Cara Sue <carasue_skx@outlook.com>; Author <author@noreply.github.com>
主题: Re: [Shpota/goxygen] fix: fix image build bug and add some note (#21)


@Shpota requested changes on this pull request.

________________________________

In templates/docker-compose.yml<https://github.com/Shpota/goxygen/pull/21#discussion_r383025812>:

> @@ -1,4 +1,4 @@
-version: "3.7"
+version: "3.2"


Why do you decrease the version of dicker compose?

________________________________

In templates/Dockerfile<https://github.com/Shpota/goxygen/pull/21#discussion_r383026035>:

>  RUN npm install && npm run build --prod

 FROM golang:1.13.8-alpine AS GO_BUILD
-RUN apk add build-base
+RUN  apk update && apk add --virtual build-dependencies build-base wget


Why do you add two additional packages if build-base is enough? Does it not work in some cases?

What would be the benefit of using --virtual here?

―

**Comment by Shpota** [comment]
Could you please tell me what OS you use, what your docker and docker compose versions are? I will try to reproduce the issue. I hope it won't require downgrading the compose version.

File: templates/docker-compose.yml (hunk 1 above)
Comment (carasue) [reply templates/docker-compose.yml]: MacOS
```
$ docker-compose version
docker-compose version 1.17.1, build 6d101fb
docker-py version: 2.5.1
CPython version: 2.7.12
OpenSSL version: OpenSSL 1.0.2j  26 Sep 2016
```
u can check here [Compose file versions and upgrading
](https://docs.docker.com/compose/compose-file/compose-versioning/)

File: templates/Dockerfile (hunk 2 above)
Comment (carasue) [reply templates/Dockerfile]: build-base is not enough for my case.
```
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.11/main: temporary error (try again later)
WARNING: Ignoring APKINDEX.70f61090.tar.gz: No such file or directory
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.11/community: temporary error (try again later)
WARNING: Ignoring APKINDEX.ca2fea5b.tar.gz: No such file or directory
2 errors; 15 distinct packages available
ERROR: Service 'app' failed to build: The command '/bin/sh -c apk update && apk add  build-base' returned a non-zero code: 2
```
And I googled, found the solution here.
```
RUN  apk update && apk add --virtual build-dependencies build-base
```
I check again, the wget package is redundant, my mistake

File: templates/docker-compose.yml (hunk 1 above)
Comment (Shpota) [reply templates/docker-compose.yml]: Version 1.17 was introduced more than 2 years ago. It supports up to v3.4 of the `docker-compose.yml`.

Could you please update Docker and Docker compose to the latest versions and check if it helps?

Comment (Shpota) [reply templates/docker-compose.yml]: Did you have a chance to check the build with updated Docker?

**Comment by nacimgoura** [comment]
On my computer (ubuntu 18.10), everything works without having to change the docker-composes file

**Comment by Shpota** [comment]
Closing as the root cause is the old version of `docker-compose`.

**Pull request closed without merging**